src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,297 @@
+package com.minecolonies.coremod.inventory.api;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagInt;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.IWorldNameable;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+
+/**
+ * Abstract class wrapping around multiple IItemHandler.
+ */
+public abstract class AbstractCombinedItemHandler
+  implements IItemHandlerModifiable, INBTSerializable<NBTTagCompound>, IWorldNameableModifiable
+{
+
+    ///NBT Constants
+    private static final String NBT_KEY_HANDLERS           = "Handlers";
+    private static final String NBT_KEY_HANDLERS_INDEXLIST = "Index";
+    private static final String NBT_KEY_NAME               = "Name";
+
+    private final IItemHandlerModifiable[] handlers;
+    @NotNull
+    private final String                   defaultName;
+    @Nullable
+    private       String                   customName;
+
+    /**
+     * Method to create a new {@link AbstractCombinedItemHandler}.
+     *
+     * @param defaultName The default name of this {@link AbstractCombinedItemHandler}.
+     * @param handlers    The combining {@link IItemHandlerModifiable}.
+     */
+    protected AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final IItemHandlerModifiable... handlers)
+    {
+        this.handlers = handlers;
+        this.defaultName = defaultName;
+    }
+
+    /**
+     * Method to create a new combined {@link AbstractCombinedItemHandler} with
+     * a given custom name.
+     *
+     * @param defaultName The name of this {@link AbstractCombinedItemHandler}.
+     * @param customName  The preset custom name of this {@link
+     *                    AbstractCombinedItemHandler}.
+     * @param handlers    The combinging {@link IItemHandlerModifiable}.
+     */
+    protected AbstractCombinedItemHandler(@NotNull final String defaultName, @NotNull final String customName, @NotNull final IItemHandlerModifiable... handlers)
+    {
+        this.handlers = handlers;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Store a copy of "handlers". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2384)